src/sdk/src/Layout/redist/targets/GenerateLayout.targets

```diff
@@ -65,6 +65,9 @@
                    AppBinaryName="%(_RoslynAppHost.Filename)%(_RoslynAppHost.Extension)"
                    IntermediateAssembly="%(_RoslynAppHost.FullPath)"
                    EnableMacOSCodeSign="$(SharedFrameworkRid.StartsWith('osx'))" />
+    
+    <Exec Command="codesign --sign - --entitlements '$(MSBuildProjectDirectory)/roslyn-entitlements.plist' %(_RoslynAppHost.RootDir)%(_RoslynAppHost.Directory)%(_RoslynAppHost.Filename)$(ExeExtension)"
+          Condition="$(SharedFrameworkRid.StartsWith('osx'))" />
```

Comment:
How do other apphosts (like `dotnet.exe`) get the entitlements? Shouldn't `CreateAppHost` task take care of that?

Reply:
we add them during the dotnet/runtime build: https://github.com/dotnet/runtime/blob/9bcdd873eb4b93934ae34328845599f63253d32a/src/native/corehost/dotnet/CMakeLists.txt#L28-L30

but I just noticed this is missing the `-f`/`--force` option and the official build fails due to that

Reply:
https://github.com/dotnet/dotnet/pull/3069

Reply:
> Shouldn't CreateAppHost task take care of that?

Ideally yes (https://github.com/dotnet/runtime/issues/113707), that's why it worked before, but we ran into issues:  https://github.com/dotnet/runtime/pull/119824

Reply:
> Ideally yes ([dotnet/runtime#113707](https://github.com/dotnet/runtime/issues/113707)), that's why it worked before, but we ran into issues: [dotnet/runtime#119824](https://github.com/dotnet/runtime/pull/119824)

I see, thanks!